common/models/templates/src/trie-model.ts

```diff
@@ -100,13 +100,62 @@ class Traversal implements LexiconTraversal {
    */
   totalWeight: number;
 
-  constructor(root: Node, prefix: string, maxWeight: number) {
+  constructor(root: Node, prefix: string, totalWeight: number) {
     this.root = root;
     this.prefix = prefix;
-    this.totalWeight = maxWeight;
+    this.totalWeight = totalWeight;
```

Comment:
I wonder if we can `Object.assign(this, {root, prefix, totalWeight});`? Syntactic prettiness only... dunno if there are downsides?

Reply:
TypeScript doesn't seem to perform type-checking well when using the suggested pattern.  I feel that this downside is too significant, so I'd rather not change to using it here.